keps/sig-node/3857-rro-mounts/README.md

```diff
@@ -0,0 +1,1080 @@
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+Constraints: needs runc >= 1.1 && kernel >= 5.12.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+Risk: a manifest with `recursivelyReadOnly: Enabled` will not work on old distros (See "Constraints" above).
+
+Mitigation: set `recursivelyReadOnly: IfPossible`.
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+
+### Core API
+Add `RecursivelyReadOnly: (Disabled|IfPossible|Enabled)` to the [`VolumeMount`](https://github.com/kubernetes/kubernetes/blob/v1.26.1/pkg/apis/core/types.go#L1854-L1880) struct:
+- `Disabled`: always non-recursive
+- `IfPossible`: recursive if possible, otherwise non-recursive, without raising an error.
+   This value is translated to `Enabled` or `Disabled` in the `VolumeMountStatus` struct (see the "Diff" below).
+- `Enabled`: recursive if possible, otherwise raise an error (as long as the node components are aware of this feature).
+
+When `RecursivelyReadOnly` is set to any non-nil value, `ReadOnly` is validated to be `true`, and if `ReadOnly` is not specified,
+the default value of `ReadOnly` will be `true`.
+
+When `RecursivelyReadOnly` is nil, `ReadOnly` is not validated to be `true`, and if `ReadOnly` is set to `true`,
+it will be a non-recursive readonly mount.
+
+When a user sets `RecursivelyReadOnly` to `IfPossible` or `Enabled`, they should also set `MountPropagation` to `"None"` (default).
+Otherwise, a writable submount may be propagated from the host mount namespace.
+
+A pod manifest will look like this:
+```yaml
+spec:
+  volumes:
+    - name: foo
+      hostPath:
+        path: /mnt
+        type: Directory
+  containers:
+  - volumeMounts:
+    - mountPath: /mnt
+      name: foo
+      mountPropagation: None
+      # NEW
+      recursivelyReadOnly: IfPossible
+```
+
+Diff:
+```diff
+diff --git a/pkg/apis/core/types.go b/pkg/apis/core/types.go
+index 04ffd250b61..b014519b626 100644
+--- a/pkg/apis/core/types.go
++++ b/pkg/apis/core/types.go
+@@ -1855,9 +1855,16 @@ type ContainerPort struct {
+ type VolumeMount struct {
+ 	// Required: This must match the Name of a Volume [above].
+ 	Name string
+-	// Optional: Defaults to false (read-write).
++	// Optional: Defaults to false (read-write), unless RecursivelyReadOnly is set to a non-nil value.
```

Comment:
Would be good to have matching comments here on the defaults when `RecursivelyReadOnly` is set.

Reply:
Added